### ISSUE TITLE
Relations generated differ between Postgres and Sqlite

### ISSUE THREAD

**Issue by addap** [SeaQL/sea-orm#1393]
## Description

Not sure if this is a bug or a limitation in Sqlite but I am trying to define a one-to-one relationship between two tables but `sea-orm-cli` always generates a `has_many` relationship. If I use Postgres instead, it actually generates a `has_one` relationship.
As I understand the comments in issue [#11]( https://github.com/SeaQL/sea-orm/issues/11#issuecomment-864403463) it should be enough to define the foreign key as unique 

## Steps to Reproduce

I am using the default migration with a second table that has a unique foreign key to the first table.
```rust
use sea_orm_migration::prelude::*;

#[derive(DeriveMigrationName)]
pub struct Migration;

#[async_trait::async_trait]
impl MigrationTrait for Migration {
    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        // Replace the sample below with your own migration scripts
        manager
            .create_table(
                Table::create()
                    .table(Post::Table)
                    .if_not_exists()
                    .col(
                        ColumnDef::new(Post::Id)
                            .integer()
                            .not_null()
                            .auto_increment()
                            .primary_key(),
                    )
                    .to_owned(),
            )
            .await?;

        manager
            .create_table(
                Table::create()
                    .table(Post2::Table)
                    .if_not_exists()
                    .col(
                        ColumnDef::new(Post2::Id)
                            .integer()
                            .not_null()
                            .auto_increment()
                            .primary_key(),
                    )
                    .col(
                        ColumnDef::new(Post2::PostId)
                            .integer()
                            .not_null()
                            .unique_key(),
                    )
                    .foreign_key(
                        ForeignKey::create()
                            .name("fk-testposts")
                            .from(Post2::Table, Post2::PostId)
                            .to(Post::Table, Post::Id),
                    )
                    .to_owned(),
            )
            .await?;

        Ok(())
    }

    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        Ok(())
    }
}

/// Learn more at https://docs.rs/sea-query#iden
#[derive(Iden)]
enum Post {
    Table,
    Id,
}

#[derive(Iden)]
enum Post2 {
    Table,
    Id,
    PostId,
}
```
Apply the migrations and generate the entities
```bash
$ DATABASE_URL="sqlite://./db.sqlite3" sea-orm-cli migrate fresh
$ sea-orm-cli generate entity -u "sqlite://./db.sqlite3" -o src/entities_sqlite
$ # switch to sqlx-postgres feature
$ DATABASE_URL="postgres://postgres:pw@127.0.0.1:5432/postgres" sea-orm-cli migrate fresh
$ sea-orm-cli generate entity -u "postgres://postgres:pw@127.0.0.1:5432/postgres" -o src/entities_pg
```

### Expected Behavior
Both times I would expect a `has_one` relation.

### Actual Behavior
In the case of Sqlite I get the following relation for `Post`.
```rust
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::post2::Entity")]
    Post2,
}
```
And the following when I use postgres
```rust
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_one = "super::post2::Entity")]
    Post2,
}
```
### Reproduces How Often
Always

## Versions
- Linux 5.19.9-arch1-1
- Postgres 15.1
- sqlite3 3.40.1
```
$ cargo tree | rg sea-                                                                                                                                          
│   └── sea-orm v0.10.6
│       ├── sea-orm-macros v0.10.6 (proc-macro)
│       ├── sea-query v0.27.2
│       │   ├── sea-query-derive v0.2.0 (proc-macro)
│       ├── sea-query-binder v0.2.2
│       │   ├── sea-query v0.27.2 (*)
│       ├── sea-strum v0.23.0
│       │   └── sea-strum_macros v0.23.0 (proc-macro)
│   └── sea-orm-migration v0.10.6
│       ├── sea-orm v0.10.6 (*)
│       ├── sea-orm-cli v0.10.6
│       │   ├── sea-schema v0.10.3
│       │   │   ├── sea-query v0.27.2 (*)
│       │   │   └── sea-schema-derive v0.1.0 (proc-macro)
│       ├── sea-schema v0.10.3 (*)
├── sea-orm v0.10.6 (*)
```

## Additional Information

The sqlite schema does contain the information that `post_id` is a foreign key and unique.
```
sqlite> .schema
CREATE TABLE sqlite_sequence(name,seq);
CREATE TABLE IF NOT EXISTS "seaql_migrations" ( "version" text NOT NULL PRIMARY KEY, "applied_at" integer NOT NULL );
...
CREATE TABLE IF NOT EXISTS "post" ( "id" integer NOT NULL PRIMARY KEY AUTOINCREMENT );
CREATE TABLE IF NOT EXISTS "post2" ( "id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "post_id" integer NOT NULL UNIQUE, FOREIGN KEY ("post_id") REFERENCES "post" ("id") );
```

**Comment by billy1624** [comment]
Hey @addap, welcome to SeaQL! Nice observation. I think we can update this part and check if it's primary key or not.

https://github.com/SeaQL/sea-orm/blob/ca205543a37ba734a90fed8f23daf1cf0e45ce0d/sea-orm-codegen/src/entity/transformer.rs#L155-L171

**Comment by addap** [comment]
Hi Billy!
Sorry, how do you mean that? Since "post_id" is not a primary key.

The `col.unique` apparently depends on whether there is an index with that column.
https://github.com/SeaQL/sea-orm/blob/ca205543a37ba734a90fed8f23daf1cf0e45ce0d/sea-orm-codegen/src/entity/transformer.rs#L51-L58

Sqlite supports indices and even creates one for `post_id`
```
sqlite> .indexes
sqlite_autoindex_post2_1             sqlite_autoindex_seaql_migrations_1
sqlite> PRAGMA index_info('sqlite_autoindex_post2_1');
0|1|post_id
```
In the discovery module there is even a function to discover the Sqlite indices (though it seems it's not called during codegen and it crashes when I call it explicitly)
https://github.com/SeaQL/sea-schema/blob/master/src/sqlite/discovery.rs#L43
To keep in line with the current design, maybe this function should be used?
However, I don't understand yet why the indices are used at all. Don't the columns in the schema have a unique annotation anyways?

**Comment by addap** [comment]
Ok to answer my own question, Sqlite automatically creates indexes for any column marked as unique.
This automatic index creation also turned out to be the problem. To get the name of the unique column, `sea-schema` was tring to parse the sql statement for the index creation which is saved in the `sql_master` table, like the following.
```
sqlite> select * from sqlite_master where name = 'myidx';
index|myidx|post2|9|CREATE UNIQUE INDEX myidx ON post2(post_id)
sqlite> select * from sqlite_master where name = 'sqlite_autoindex_post2_1';
index|sqlite_autoindex_post2_1|post2|8|

```
However, the last column is empty for automatically created indices. I changed it so that it uses the `PRAGMA index_info` command to get the column names and now the correct attribute `#[sea_orm(has_one = "super::post2::Entity")]` is generated. Btw does this attribute even do anything? I did not find where it is used.

I will create a pull request for this in the sea-schema repo.
